### PR TITLE
Egress DNS Policy changes

### PR DESCRIPTION
**Pull request by shastrinator**
DNS resolution using datapath response parsing
Add a new table just before OUT table in access bridge to
tap DNS response packets.

TODO:
DNS expiry
IPv6 and CName record processing
Dynamic DNS punting
Prepopulate DNS cache with k8s services

Signed-off-by: Kiran Shastri <shastrinator@gmail.com>

### CODE REVIEW

File: genie/MODEL/SPECIFIC/EPR/epdr.mdl [hunk 1]
```diff
@@ -180,4 +180,88 @@ module[epdr]
...
+    class[DnsEntry;
+          super=policy/NamedDefinition;
+          concrete]
+    {
+        member[updated; type=ascii/String]
+        member[expiry; type=scalar/UInt32]
+        dependency[mapped;
+                   cardinality=many;]
+        {
+            to[class=epdr/DnsMappedAddress;
```
Comment (tomflynn) [inline genie/MODEL/SPECIFIC/EPR/epdr.mdl:211]: DnsMappedAddress is contained in DnsEntry. Why do you need to have a relation from DnsEntry to DnsMappedAddress as well?

Comment (shastrinator) [reply genie/MODEL/SPECIFIC/EPR/epdr.mdl:211]: ok missed removing this.. will do.. thanks.

**Review by tomflynn — APPROVED** [review]
Just some minor nits now that the model looks good

File: agent-ovs/lib/include/opflexagent/PolicyManager.h [hunk 2]
```diff
@@ -831,7 +842,18 @@ class PolicyManager : private boost::noncopyable {
...
+     * Type to hold a set of (DNS) names
+     */
+    typedef std::unordered_set<std::string> named_addr_set_t;
+    /**
+     * Get cached Dns resolved addresses for a given domain name.
+     * if they exist in the cache. Call with the state mutex held.
+     * @param domainName
+     * @param egressDnsResolved resolved address set
+     */
+    void getDnsResolvedAddress( const std::string &domainName,
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply agent-ovs/lib/include/opflexagent/PolicyManager.h:855]: Sure

File: agent-ovs/lib/PolicyManager.cpp [hunk 3]
```diff
@@ -1090,12 +1124,122 @@ void sortOrderOfSameRange(vector<shared_ptr<modelgbp::gbpe::L24Classifier>>& cla
...
+    using modelgbp::epdr::DnsMappedAddress;
+    using network::operator<<;
+
+    if(class_id != DnsAnswer::CLASS_ID) {
+        return;
+    }
+    auto dnsAnswer = DnsAnswer::resolve(framework, uri);
+    if(!dnsAnswer) {
+        vector<string> elements;
+	uri.getElements(elements);
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply agent-ovs/lib/PolicyManager.cpp:1141]: some tabs instead of spaces..fixed them.

**Review by tomflynn — APPROVED** [review]

**Comment by shastrinator** [comment]
Fixed some more tab issues

**Pull request merged**